src/every.ts

```diff
@@ -0,0 +1,94 @@
+import type {
+  Pred,
+  PredIndexed,
+  PredIndexedOptional,
+  TypePred,
+  TypePredIndexed,
+} from "./_types";
+import { purry } from "./purry";
+
+/**
+ * Checks if every element of `data` passed the predicate `predicate`.
+ *
+ * @param data - The Array to check.
+ * @param predicate - A predicate function to run.
+ * @returns A `boolean` indicating if every element in `data` passes the predicate `pred`.
+ * @signature
+ *    R.every(arr, predicate)
+ * @example
+ *    R.every([1, 2, 3], R.isNumber) // => true
+ *    R.every(['a', 0, 'b'], R.isString) // => false
+ * @dataFirst
+ * @indexed
+ * @category Array
+ */
+export function every<T, S extends T>(
+  data: ReadonlyArray<T>,
+  predicate: TypePred<T, S>,
+): data is Array<S>;
+export function every<T>(
+  data: ReadonlyArray<T>,
+  predicate: Pred<T, boolean>,
+): boolean;
+
+/**
+ * Checks if every element of `data` passed the predicate `predicate`.
+ *
+ * @param predicate - A predicate function to run.
+ * @returns A `boolean` indicating if every element in `data` passes the predicate `pred`.
+ * @signature
+ *    R.every(predicate)(data)
+ * @example
+ *    R.every(isNumber)([1, 2, 3]) // => true
+ *    R.every(isString)(['a', 0, 'b']) // => false
+ * @example
+ *    // using `R.every` as a type predicate within `conditional`
+ *    const allNumbers = R.every(R.isNumber);
+ *    const result = R.pipe(
+ *      R.conditional(
+ *        [allNumbers, val => {...}]
+ *                  // ^? number[]
+ *        ...
+ *      )
+ *    )
```

Comment:
I don't know if we support multiple example tags in our docs. This seems like a really big example using a utility that people might not know or understand; it makes the example more complex than it should be.

Our examples aren't usually full-blown usage examples; they just show a basic, almost trivial, use case so that people understand what the params look like and how they relate to the output.